exporters/ostream/include/opentelemetry/exporters/ostream/span_exporter.h

```diff
@@ -89,6 +89,9 @@ class OStreamSpanExporter final : public sdktrace::SpanExporter
     sout_ << ']';
   }
 
+// Prior to C++14, generic lambda is not available so fallback to functor.
+#if __cplusplus < 201402L
```

Comment:
LGTM in general. Although I had concern with this macro whether all the compilers define it properly ( I know there were issues earlier, probably not any more ).
jfyi - there are also feature specific recommendation - eg  `__cpp_generic_lambdas` for compilers supporting generic lambda.  Ref: https://isocpp.org/std/standing-documents/sd-6-sg10-feature-test-recommendations#__cpp_generic_lambdas